configure.py

```diff
@@ -60,55 +71,53 @@ def main(args):
     #llvm_projects.append("iree_dialects")
     llvm_configure_args.append(
         f"-DLLVM_EXTERNAL_IREE_DIALECTS_SOURCE_DIR={iree_dialects_path}")
-
-  # Detect LLVM.
-  if args.llvm_path:
-    llvm_path = os.path.abspath(args.llvm_path)
-    print(f"-- Using explicit llvm-project path: {llvm_path}")
-  elif llvm_path:
-    print(f"-- Using inferred llvm-project path: {llvm_path}")
   else:
-    llvm_path = os.path.join(args.repo_root, "..", "llvm-project")
+    # Detect LLVM.
+    # As per "Building without IREE" instructions:
+    # You **must** checkout [llvm-project](https://github.com/llvm/llvm-project) at a
+    # compatible commit next to this directory.
+    llvm_path = os.path.join(repo_root, "..", "llvm-project")
     print(f"-- Using default llvm-project path: {llvm_path}")
-  if not os.path.exists(llvm_path):
-    print(f"ERROR: Could not find llvm-project at {llvm_path}")
-    return 1
+    if not os.path.exists(llvm_path):
+      print(f"ERROR: Could not find llvm-project at {llvm_path}")
+      return 1
 
   # Detect clang.
   clang_path = shutil.which("clang")
   clangpp_path = shutil.which("clang++")
-  has_clang = False
   if clang_path and clangpp_path:
     llvm_configure_args.append(f"-DCMAKE_C_COMPILER={clang_path}")
     llvm_configure_args.append(f"-DCMAKE_CXX_COMPILER={clangpp_path}")
-    has_clang = True
   else:
     print(
         "WARNING: Could not find clang. Building with default system compiler")
 
   # Detect lld.
-  lld_path = shutil.which("ld.lld")
-  if not lld_path:
-    print("WARNING: LLD (ld.lld) not found on path. Configure may fail.")
+  if args.enable_lld:
+    lld_path = shutil.which("ld.lld")
+    if lld_path:
+      print(f"-- Using lld: {lld_path}")
+      llvm_configure_args.append("-DLLVM_ENABLE_LLD=ON")
+    else:
+      print("WARNING: LLD (ld.lld) not found on path. Configure may fail.")
 
   # Detect ccache.
-  ccache_path = shutil.which("ccache")
   if args.enable_ccache:
+    ccache_path = shutil.which("ccache")
     if ccache_path:
-      print(f"-- Using ccache")
+      print(f"-- Using ccache: {ccache_path}")
       llvm_configure_args.append("-DLLVM_CCACHE_BUILD=ON")
     else:
       print("WARNING: Project developers use ccache which is not installed")
 
   # CMake configure.
-  build_dir = os.path.abspath(args.build_dir)
+  build_dir = os.path.abspath(os.getenv("IREE_LLVM_SANDBOX_BUILD_DIR"))
   os.makedirs(build_dir, exist_ok=True)
   cmake_args = [
       "cmake",
       "-GNinja",
       f"-B{build_dir}",
       f"-S{os.path.join(llvm_path, 'llvm')}",
-      "-DLLVM_ENABLE_LLD=ON",
       f"-DLLVM_ENABLE_PROJECTS={';'.join(llvm_builtin_projects)}",
```

Comment:
this was already here before and it works for me; does it break for you?

Reply:
I think it gets escaped properly down the line by subprocess. Generally, you can count on each argument being an actual argument in the executable argv

Reply:
I was manually adding quotes before :). To me it breaks with: 
```
/bin/sh: clang: command not found
/bin/sh: clang-tools-extra: command not found
```
Might be because of a different shell?

Reply:
Weird. If the quotes help and don't break the CI, fine with me (I've seen cases where adding quotes to this API does break things).